spinalcordtoolbox/scripts/sct_get_centerline.py

```diff
@@ -71,6 +71,14 @@ def get_parser():
         default=30,
         help="Degree of smoothing for centerline fitting. Only for -centerline-algo {bspline, linear}."
     )
+    optional.add_argument(
+        "-centerline-soft",
+        metavar=Metavar.int,
+        type=int,
+        choices=[0, 1],
+        default=0,
+        help="Binary or soft centerline. 0 = binarized, 1 = soft. Only relevant with -method fitseg."
+    )
```

Comment:
Sorry to suggest one more change, but. Could you write a test for this new option?

It would be nice to run the command twice (`-centerline-soft {0,1}`) then compare the outputs to make sure:

- The softseg output is equal to 1 on all slices.
- The softseg overlaps with the binary segmentation on all slices.

This might be a little overkill? But it would allow us to make changes to this code in the future with confidence.

Reply:
Sure, great suggestion! I will write tests next week.

Reply:
Hey @joshuacwnewton! @NathanMolinier and I tried to implement some first tests in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4026/commits/e8b25c9921e24b721b010ecd245e87e978e0e989 and https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4026/commits/120c94bc3f0414e0b4093816babb3079d271e374. It is not the final version yet, we just wanted to check if CI passes. But the CI has not started; instead, it is: "Waiting for status to be reported". Could you please restart it?
Also, there are some conflicts with master, could you please resolve them? Thank you!

Reply:
I believe the two problems are one and the same! (I think the conflicts need to be resolved for the CI to run.)

I'll take care of this right now. :)

Reply:
Note: The reason for the conflict was because of #4031, which removes the `round_and_clip` function. Resolving the conflict is as easy as replacing the `round_and_clip` parts of your PR with the version on master:

![image](https://user-images.githubusercontent.com/16181459/219816110-8418e261-e635-43af-ac2b-e11f14564a46.png)